flink-table/flink-table-planner/src/main/scala/org/apache/flink/table/planner/plan/metadata/FlinkRelMdColumnUniqueness.scala

```diff
@@ -81,9 +80,22 @@ class FlinkRelMdColumnUniqueness private extends MetadataHandler[BuiltInMetadata
       return false
     }
 
-    // TODO get uniqueKeys from TableSchema of TableSource
-
     relOptTable match {
+      case sourceTable: TableSourceTable =>
```

Comment:
from your change, I have another idea: we can get the unique keys first in a unified way via `mq.getUniqueKeys(rel)` and then check whether the given columns are unique. WDYT ?

Reply:
Thx, Please check again.